main.tf

```diff
@@ -336,13 +339,13 @@ module "rds_identifier" {
 resource "aws_rds_cluster_instance" "default" {
   count                                 = local.cluster_instance_count
   identifier                            = "${module.rds_identifier[0].id}-${count.index + 1}"
-  cluster_identifier                    = coalesce(join("", aws_rds_cluster.primary[*].id), join("", aws_rds_cluster.secondary[*].id))
+  cluster_identifier                    = random_pet.instance[0].keepers.cluster_identifier
```

Comment:
this is forcing a replacement in the `resource "aws_rds_cluster_instance" "default" {` now that these attribute values are changing (even if the end result value is the same). this would cause a recreate of the database instance... see example plan above

Reply:
@aknysh